toolkit/tools/srpmpacker/srpmpacker.go

```diff
@@ -273,16 +460,18 @@ func specsToPackWorker(allSpecFiles chan string, specResults chan *specState, di
 		if err != nil {
 			if err.Error() == rpm.NoCompatibleArchError {
 				logger.Log.Infof("Skipping SPEC (%s) due to incompatible build architecture", specFile)
-				specResults <- result
-				continue
 			} else {
-				// On error, the `rpm` package will automatically log the stderr to the `warn` level.
-				logger.Log.Fatalf("Failed to query SPEC (%s), error: %s", specFile, err)
+				result.err = err
```

Comment:
>result.err = err [](start = 4, length = 16)

Why not just return immediately here instead of continuing below?

Reply:
In this pattern (present in a few tools now), it is up to the routine orchestrating the worker pool to decide what is fatal and what is not. Having this pattern easily allows us to change the tooling to pack/parse best effort and not immediately fail in the future.

Example: `STOP_ON_PKG_FAIL` configures the scheduler to stop the worker routines on any error during build. Leaving this decision up to one routine easily enables these kinds of arguments.